fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryUnorderedDistinctPlan.java

```diff
@@ -96,15 +95,15 @@ private RecordQueryUnorderedDistinctPlan(@Nonnull final Quantifier.Physical inne
 
     @Nonnull
     @Override
-    public <M extends Message> RecordCursor<FDBQueriedRecord<M>> execute(@Nonnull FDBRecordStoreBase<M> store,
-                                                                         @Nonnull EvaluationContext context,
-                                                                         @Nullable byte[] continuation,
-                                                                         @Nonnull ExecuteProperties executeProperties) {
+    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
+                                                                     @Nonnull final EvaluationContext context,
+                                                                     @Nullable final byte[] continuation,
+                                                                     @Nonnull final ExecuteProperties executeProperties) {
         final Set<Key.Evaluated> seen = new HashSet<>();
-        return getInner().execute(store, context, continuation, executeProperties.clearSkipAndLimit())
-            .filterInstrumented(record -> seen.add(getComparisonKey().evaluateSingleton(record)),
-                store.getTimer(), Collections.emptySet(), duringEvents, uniqueCounts, duplicateCounts)
-            .skipThenLimit(executeProperties.getSkip(), executeProperties.getReturnedRowLimit());
+        return getInner().executePlan(store, context, continuation, executeProperties.clearSkipAndLimit())
+                .filterInstrumented(result -> seen.add(getComparisonKey().evaluateSingleton(result.getQueriedRecord(0))),
```

Comment:
This branch does not change the semantics, but looking at it I wonder whether it's intentional that a skipped record goes into `seen`, meaning that it won't be output should it occur again after the skip.

Reply:
This PR should not have changed the behavior, and no tests have failed. Should I open a ticket?

Reply:
Yes, I think we need to track this.

Reply:
rdar://79577753 (RecordQueryUnorderedDistinctPlan "sees" a record when it is skipped)